R/req-promise.R

```diff
@@ -131,12 +131,20 @@ ensure_pool_poller <- function(pool, reject) {
   monitor <- pool_poller_monitor(pool)
   if (monitor$already_going()) return()
 
-  poll_pool <- function() {
+  poll_pool <- function(ready) {
     tryCatch(
       {
         status <- curl::multi_run(0, pool = pool)
         if (status$pending > 0) {
-          later::later(poll_pool, delay = 0.1, loop = later::global_loop())
+          fds <- curl::multi_fdset(pool = pool)
+          later::later_fd(
+            func = poll_pool,
+            readfds = fds$reads,
+            writefds = fds$writes,
+            exceptfds = fds$exceptions,
+            timeout = fds$timeout,
+            loop = later::global_loop()
```

Comment:
I think I was just adapting from https://github.com/rstudio/promises/blob/main/R/future_promise.R#L153 

Not sure if that discussion is relevant here since the work is submitted to curl, but also seemed like I might be opening a can of worms by not using the global loop.

Reply:
Ah, OK. I'm 99.9% sure it's better to use the current loop here (but 50/50 on what's better for future_promise). @shikokuchuo if you make me a collaborator on your httr2 fork I'll push a unit test for this case.

Reply:
Yes, I'd just come to the same conclusion (for here). Amended in [8078638](https://github.com/shikokuchuo/httr2/pull/1/commits/807863847150ed71cdba66d90b29ce5ac532b7a6).